### PR TITLE
Ensures mesh name is compatible with USD convention in mesh converter

### LINKED ISSUE #1287: [Bug Report] `convert_mesh.py` failed when `*.obj` file name starts with a number
### Describe the bug

When using the `convert_mesh.py` script to convert `.obj` to `.usd`, if the obj file in this scene starts with a number(`0c334eaabb844eaaad049cbbb2e0a4f2.obj`), while the USD API prohibits names that start with a number [Link](https://openusd.org/release/api/group__group__tf___string.html#gaa129b294af3f68d01477d430b70d40c8). This issue is described in more detail here: [Link](https://forums.developer.nvidia.com/t/cant-create-prims-if-name-starts-with-number/249617). 

```bash
Usd.Stage.Open(rootLayer=Sdf.Find('anon:0x23b593e0:tmp.usda'), sessionLayer=Sdf.Find('anon:0x23b3f000:tmp-session.usda'), pathResolverContext=<invalid repr>)
2024-10-22 11:14:00 [5,398ms] [Warning] [omni.usd] Warning: in SdfPath at line 81 of /builds/omniverse/usd-ci/USD/pxr/usd/sdf/path.cpp -- Ill-formed SdfPath </0c334eaabb844eaaad049cbbb2e0a4f2>: syntax error

Traceback (most recent call last):
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/../IsaacLab/source/standalone/tools/convert_mesh.py", line 177, in <module>
    main()
  File "/home/ubuntu/workspaces/viplanner_ws/viplanner/../IsaacLab/source/standalone/tools/convert_mesh.py", line 147, in main
    mesh_converter = MeshConverter(mesh_converter_cfg)
  File "/home/ubuntu/workspaces/viplanner_ws/IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py", line 58, in __init__
    super().__init__(cfg=cfg)
  File "/home/ubuntu/workspaces/viplanner_ws/IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/asset_converter_base.py", line 104, in __init__
    self._convert_asset(cfg)
  File "/home/ubuntu/workspaces/viplanner_ws/IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py", line 85, in _convert_asset
    asyncio.get_event_loop().run_until_complete(
  File "/home/ubuntu/anaconda3/envs/viplanner/lib/python3.10/site-packages/omni/extscore/omni.kit.async_engine/omni/kit/async_engine/async_engine.py", line 160, in run_until_complete
    return self.super_run_until_complete(future)
  File "/home/ubuntu/anaconda3/envs/viplanner/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/ubuntu/workspaces/viplanner_ws/IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py", line 223, in _convert_mesh_to_usd
    base_prim = temp_stage.DefinePrim(prim_path, "Xform")
pxr.Tf.ErrorException: 
	Error in 'pxrInternal_v0_22__pxrReserved__::UsdStage::_IsValidPathForCreatingPrim' at line 3340 in file /builds/omniverse/usd-ci/USD/pxr/usd/usd/stage.cpp : 'Path must be an absolute path: <>'
```

Perhaps we should modify the `_convert_asset()` function in `convert_mesh.py` script to throw a user-friendly warning when the `mesh_file_basename` does not comply with the USD file format.

This issue was first encountered in https://github.com/leggedrobotics/viplanner/pull/32#issuecomment-2429031132

### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### PR DESCRIPTION
**Pull request by fan-ziqi**
# Description

Fixes #1287

**Issue details**: When using the convert_mesh.py script to convert .obj to .usd, if the obj file in this scene starts with a number(0c334eaabb844eaaad049cbbb2e0a4f2.obj), while the USD API prohibits names that start with a number [Link](https://openusd.org/release/api/group__group__tf___string.html#gaa129b294af3f68d01477d430b70d40c8). This issue is described in more detail here: [Link](https://forums.developer.nvidia.com/t/cant-create-prims-if-name-starts-with-number/249617).

Also deleted useless debugging output code.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

Please attach before and after screenshots of the change if applicable.

| Before | After |
| ------ | ----- |
| old | new |

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Review by Dhoeller19 — APPROVED** [review]
Thanks @fan-ziqi

**Comment by fan-ziqi** [comment]
I just simplified the processing logic and removed useless intermediate variables

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py [hunk 1]
```diff
@@ -81,6 +81,13 @@ def _convert_asset(self, cfg: MeshConverterCfg):
         mesh_file_basename, mesh_file_format = os.path.basename(cfg.asset_path).split(".")
         mesh_file_format = mesh_file_format.lower()
 
+        # Check if mesh_file_basename is a valid USD identifier
+        if not Tf.IsValidIdentifier(mesh_file_basename):
+            print(f"Warning: File name '{mesh_file_basename}.{mesh_file_format}' is invalid for the input mesh path.")
+            # Correct the name to a valid identifier ang update the basename
+            mesh_file_basename = Tf.MakeValidIdentifier(mesh_file_basename)
+            print(f"Warning: Using '{mesh_file_basename}.{mesh_file_format}' as the identifier for this conversion.")
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py]:
```suggestion
            # Correct the name to a valid identifier and update the basename
            mesh_file_basename_original = mesh_file_basename
            mesh_file_basename = Tf.MakeValidIdentifier(mesh_file_basename)
            omni.log.warn(
                f"Input file name '{mesh_file_basename_original}' is an invalid identifier for the mesh prim path. Renaming it to '{mesh_file_basename}' for the conversion."
             )
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py [hunk 2]
```diff
@@ -81,6 +81,15 @@ def _convert_asset(self, cfg: MeshConverterCfg):
         mesh_file_basename, mesh_file_format = os.path.basename(cfg.asset_path).split(".")
         mesh_file_format = mesh_file_format.lower()
 
+        # Check if mesh_file_basename is a valid USD identifier
+        if not Tf.IsValidIdentifier(mesh_file_basename):
+            # Correct the name to a valid identifier and update the basename
+            mesh_file_basename_original = mesh_file_basename
+            mesh_file_basename = Tf.MakeValidIdentifier(mesh_file_basename)
+            omni.log.warn(
+                f"Input file name '{mesh_file_basename_original}' is an invalid identifier for the mesh prim path. Renaming it to '{mesh_file_basename}' for the conversion."
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py]:
```suggestion
                f"Input file name '{mesh_file_basename_original}' is an invalid identifier for the mesh prim path."
                f" Renaming it to '{mesh_file_basename}' for the conversion."
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py [hunk 3]
```diff
@@ -81,6 +81,16 @@ def _convert_asset(self, cfg: MeshConverterCfg):
...
 
+        # Check if mesh_file_basename is a valid USD identifier
+        if not Tf.IsValidIdentifier(mesh_file_basename):
+            # Correct the name to a valid identifier and update the basename
+            mesh_file_basename_original = mesh_file_basename
+            mesh_file_basename = Tf.MakeValidIdentifier(mesh_file_basename)
+            omni.log.warn(
+                f"Input file name '{mesh_file_basename_original}' is an invalid identifier for the mesh prim path."
+                f" Renaming it to '{mesh_file_basename}' for the conversion."
+             )
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/sim/converters/mesh_converter.py]:
```suggestion
            )
```

**Pull request merged**